### ISSUE TITLE
Beautifulsoup dependency

### ISSUE THREAD

**Issue by omangin** [pubs/pubs#31]
Beautifulsoup is a required dependency but not mentioned as such in setup.py nor readme.

Anyway I am not sure this dependency should be required and maybe it would be better as an optional one with dynamic imports?

Any idea?

**Comment by 73** [comment]
The way the software behaves now (not running when not installed) it is required (but missing from setup.py).
It might be good to add it to `install_requires`, so that people can run your code and implement a feature that makes it optional?

Best,
73

**Comment by humm** [comment]
@73 You are right. As long as the decision has not been made to make BeautifulSoup optional, it should not break the rest of the program. Added beautifulsoup4 to `install_requires`.

I am leaving the ticket open for now, to further discuss if BeautifulSoup should be required.

**Comment by 73** [comment]
There is some plugin capability there. Maybe the whole feature should be a plugin and automatically enabled if the dependencies are fulfilled?

**Comment by benureau** [comment]
@omangin: can we close this? I view ISBN import as a core feature. That would be really bad for our library to says that *books* are a secondhand or plugin feature. I looked at alternative ways to do it with library such as https://github.com/xlcnd/isbnlib, but the WorldCat API is being closed-off behind identification. Using ottolib is our only option to bypass this, as alternative options such as the Google Search API or OpenLibrary.org API don't have 9782081336742 in their database, making them insufficient for our purposes.

**Comment by omangin** [comment]
Yes. It has been fixed, since `beautifulsoup` is now a dependency.

We had this debate about limiting the number of dependencies by making some functionalities depend on optional ones. We decided instead to make `beautifulsoup` and `feedparser` dependencies. These are pretty standard libraries and that way we have the functionalities in the core of `pubs`. If in the future we start importing more _exotic_ libraries for other APIs we might consider making them optional (by putting the functionality in a plugin or simply by failing accordingly).

>  I view ISBN import as a core feature. That would be really bad for our library to says that books are a secondhand or plugin feature.

I am not sure that making ISBN import optional would really make books second hand in `pubs` 😉
